openapi.yml

```diff
@@ -146,19 +163,69 @@ components:
       format: FiscalCode
       x-import: italia-ts-commons/lib/strings
       example: SPNDNL80R13C555X
+    ArrayIndexUpTo4:
+      type: integer
+      minimum: -1
+      maximum: 5
+      exclusiveMaximum: true
+    ExistingArrayIndexUpTo4:
+      type: integer
+      minimum: 0
+      maximum: 5
+      exclusiveMaximum: true
+    NonEmptyString:
+      type: string
+      format: NonEmptyString
```

Comment:
I'd use `minLength` or `pattern`, eg.

```
    NonEmptyString:
      type: string
      minLength: 1 
    NonEmptyString2:
      type: string
      pattern: '.+'
```

Reply:
If there's a `format registry` referencing `format` details please provide a link
in the `description`

Reply:
just want to point out thet if you use `minLength: 1` it will be automatically converted to a `NonEmptyString` typescript type by io-utils code generator

Reply:
I used the same approach which was used [here](https://github.com/teamdigitale/io-functions/blob/v0.64.0/api/definitions.yaml#L238).
If I remove the format and use `minLength: 1`, then the file generated by io-utils contains a typescript error because `NonEmptyString` name is both a name used in the import from italia-ts-commons and in the const and type definitions:
```
import { NonEmptyString } from "italia-ts-commons/lib/strings";
import * as t from "io-ts";

export type NonEmptyString = t.TypeOf<typeof NonEmptyString>;
export const NonEmptyString = NonEmptyString;
```

Reply:
That's exactly why we dont want to have another `NonEmptyString` definition here (it will conflict with the generated one). I mean, you can use `minLength: 1` directly on the properties without introducing a new `NonEmptyString` here in the specs.

Reply:
I got it now, thanks!